### PR TITLE
A series of small upgrades to automated testing scripts

### PR DESCRIPTION
**Pull request by bartgol**
Includes:

1) recycle existing baselines in jenkins jobs, unless they are missing or expired (meaning sha used to generate them does not match the sha of origin/master);
2) only build baseline executables when generating baselines;
3) always build baselines in a directory different from where the tests are built.

### CODE REVIEW

**Comment by bartgol** [comment]
Failures are due to permissions for jenkins user in the baselines folder. The easiest (and safer) way to fix this is to add jenkins to the group associated to the folder (rather than change permissions to allow everyone to write).

**Comment by jgfouca** [comment]
@bartgol , what group owns the directory?

File: components/scream/scripts/jenkins/jenkins_common.sh [hunk 1]
```diff
@@ -27,7 +27,9 @@ if [ $skip_testing -eq 0 ]; then
       SUBMIT="" # We don't submit AT runs
   fi
 
-  ./scream/components/scream/scripts/gather-all-data "./scripts/test-all-scream \$compiler -p -i -m \$machine $SUBMIT" -l -m $SCREAM_MACHINE
+  BASELINES_DIR=/home/projects/e3sm/scream/pr-autotester/master-baselines/$SCREAM_MACHINE
```
Comment (jgfouca) [inline components/scream/scripts/jenkins/jenkins_common.sh:30]: This field should be implemented via the machine dictionary. This path is not going to work for the various machines being tested outside of sandia.

Comment (bartgol) [reply components/scream/scripts/jenkins/jenkins_common.sh:30]: Good catch. I have to think about how to handle this, cause I do need to pass something to `--baseline-dir`.

**Comment by jgfouca** [comment]
We will need to keep a close eye on the nightly results tomorrow.

**Comment by bartgol** [comment]
I'm going to merge, since the last mods since Jim's approval were a python grammar error and a stranded string in a bash command.

**Pull request merged**